### PR TITLE
drivers: memc: introduce driver for APS6408L PSRAM

### PR DESCRIPTION
**Pull request by danieldegrasse**
Introduce driver for APS6408L PSRAM, present on the RT595 EVK. This driver was tested using `samples/drivers/memc`, which is introduced with this PR.

@pdgendt, FYI- I've renamed the binding and file for the HyperRAM driver to match the HyperRAM chip (S27KS0641) that this memory controller supports. From what I can tell, the driver will only support this (and similar Infineon) HyperRAM chips, although if that is not the case please feel free to clarify

### CODE REVIEW

File: samples/drivers/memc/README.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+##################
+
+Overview
+********
+
+This sample can be used with any memory controller driver that
+memory maps external RAM. It is intended to demonstrate
+the ability to read and write from the memory mapped region.
+
+Note that the chosen region (set by zephyr,sram-ext) chosen node should not
```
Comment (pdgendt) [inline samples/drivers/memc/README.rst:13]: This is an `alias` instead of a `chosen` node.

Comment (danieldegrasse) [reply samples/drivers/memc/README.rst:13]: It is, sorry- changed that right before pushing and forgot to update the documentation. Will fix that now

**Review by pdgendt — DISMISSED** [review]

File: dts/bindings/mtd/nxp,imx-flexspi-aps6408l.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Copyright 2022 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: Cypress APS6408L HyperRAM on NXP FlexSPI bus
```
Comment (DerekSnell) [inline dts/bindings/mtd/nxp,imx-flexspi-aps6408l.yaml:4]:
```suggestion
description: AP Memory APS6408L pSRAM on NXP FlexSPI bus
```

File: drivers/memc/Kconfig.mcux [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
-	bool "MCUX FlexSPI HyperRAM driver"
+config MEMC_MCUX_FLEXSPI_S27KS0641
+	bool "MCUX FlexSPI Cypress S27KS0641 HyperRAM driver"
 	default y
-	depends on DT_HAS_NXP_IMX_FLEXSPI_HYPERRAM_ENABLED
+	depends on DT_HAS_NXP_IMX_FLEXSPI_S27KS0641_ENABLED
+	select MEMC_MCUX_FLEXSPI
+
+config MEMC_MCUX_FLEXSPI_APS6408L
+	bool "MCUX FlexSPI APmemory APS6408L HyperRAM driver"
```
Comment (DerekSnell) [inline drivers/memc/Kconfig.mcux:14]:
```suggestion
	bool "MCUX FlexSPI AP Memory APS6408L pSRAM driver"
```

File: drivers/memc/memc_mcux_flexspi_aps6408l.c [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+		.data = NULL,
+		.dataSize = 0,
+	};
+
+	LOG_DBG("Resetting ram");
+	ret = memc_flexspi_transfer(data->controller, &transfer);
+	if (ret < 0) {
+		return ret;
+	}
+	/* We need to delay 5 ms to allow APS6408L HyperRAM to reinitialize */
```
Comment (DerekSnell) [inline drivers/memc/memc_mcux_flexspi_aps6408l.c:200]:
```suggestion
	/* We need to delay 5 ms to allow APS6408L pSRAM to reinitialize */
```

File: drivers/memc/memc_mcux_flexspi_aps6408l.c [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+				    (const uint32_t *) memc_flexspi_aps6408l_lut,
+				    sizeof(memc_flexspi_aps6408l_lut) / 4)) {
+		LOG_ERR("Could not update lut");
+		return -EINVAL;
+	}
+
+	memc_flexspi_reset(data->controller);
+
+	if (memc_flexspi_aps6408l_reset(dev)) {
+		LOG_ERR("Could not reset HyperRAM");
```
Comment (DerekSnell) [inline drivers/memc/memc_mcux_flexspi_aps6408l.c:233]:
```suggestion
		LOG_ERR("Could not reset pSRAM");
```

**Review by DerekSnell — CHANGES_REQUESTED** [review]
Thank you @danieldegrasse ,

Functionally this PR looks good to me.  But we do have some comments that need to be corrected.

Thanks

**Comment by danieldegrasse** [comment]
@DerekSnell Thank you for the suggestions- all instances of HyperRAM should now be changed to pSRAM for this driver, and the manufacturer has been corrected in docstrings.

**Review by DerekSnell — APPROVED** [review]

**Comment by danieldegrasse** [comment]
this PR now depends on the fix present in #53318

**Review by pdgendt — APPROVED** [review]

**Pull request merged**